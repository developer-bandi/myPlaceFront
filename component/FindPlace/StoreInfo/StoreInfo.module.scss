$commonPadding: 15px;

.mainBlock {
  width: 380px;
  height: calc(100vh - 61px);
  background-color: white;
  position: absolute;
  z-index: 25;
  box-shadow: 15px 0px 15px -10px rgba(0, 0, 0, 0.2);
  left: 380px;
  border-left: 1px solid #dddddd;
  overflow-y: scroll;
  @media screen and (max-width: 767px) {
    width: 100%;
    left: 0;
    z-index: 26;
    &.fold {
      top: calc(((100vh - 121px) / 2) + 121px);
      height: calc((100vh - 121px) / 2);
    }
    &.full {
      top: 121px;
      height: calc(100vh - 121px);
    }
  }
}

.titleBlock {
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 20px;
  font-size: 20px;
}

.storeName {
}

.moveList {
  position: absolute;
  left: 0;
  margin-left: 10px;
}
.category {
  color: #a1a1a1;
  font-size: 15px;
  margin-left: 10px;
}

.buttonBlock {
  display: flex;
  justify-content: center;
  margin-top: 10px;
}

.infoBlock {
  padding: $commonPadding;
}

.infoContentBlock {
  display: flex;
  align-items: flex-start;
  margin-bottom: 15px;
}

.icon {
  margin-right: 10px;
  color: #adadad;
}

.menuListBlock {
  padding: $commonPadding;
  display: flex;
  flex-wrap: wrap;
}

.menuBlock {
  width: 350px;
  display: flex;
  overflow-x: scroll;
}

.menuImgBlock {
  flex: 0 0 auto;
  margin-right: 10px;
}

.tagListBlock {
  display: flex;
  flex-wrap: wrap;
  margin-bottom: 10px;
}

.subTitle {
  margin-right: 250px;
  margin-bottom: 15px;
}

.tag {
  margin-right: 5px;
  margin-bottom: 5px;
  border: 1px solid #f0f0f2;
  border-radius: 10px;
  color: #666666;
  padding: 5px;
}

.tagCount {
  margin: 0px 0px 0px 5px;
}

.reviewBlock {
  padding: $commonPadding;
}

.commentBlock {
  border-bottom: 2px solid #f6f8f9;
  padding-top: 10px;
}

.commentInfo {
  display: flex;
  align-items: center;
  margin-bottom: 10px;
}

.tagListBlock {
  margin-top: 5px;
}

.commentTag {
  margin-right: 10px;
  margin-bottom: 5px;
}

.imgBlock {
  display: flex;
  margin-bottom: 10px;
}

.imgs {
  width: 100px;
  height: 100px;
  margin-right: 10px;
}

.commentContent {
  margin-bottom: 10px;
  white-space: pre-wrap;
  line-height: 1.5;
  font-size: 15px;
  color: #424242;
}

.commentTag {
  font-size: 12px;
  color: gray;
}

.contributeButton {
  width: 65px;
  height: 30px;
  border: 1px solid gray;
  border-radius: 5px;
  margin-right: 10px;
}

.bookmarkIcon {
  position: absolute;
  right: 0;
  margin-right: 10px;
}

.date {
  color: #a1a1a1;
  margin-left: 10px;
  font-size: 14px;
}

.noReview {
  text-align: center;
  font-size: 20px;
  font-weight: 600px;
}

.nickname {
  font-weight: 600;
}

.loadingBlock {
  position: relative;
  width: 380px;
  height: calc(100vh - 61px);
}

.error {
  position: relative;
  width: 380px;
  height: calc(100vh - 61px);
  font-size: 25px;
}
