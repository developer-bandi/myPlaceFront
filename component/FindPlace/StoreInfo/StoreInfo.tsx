import styles from "./StoreInfo.module.scss";
import {BsFillTelephoneFill} from "react-icons/bs";
import {GiPositionMarker} from "react-icons/gi";
import {BiTimeFive} from "react-icons/bi";
import {FaRegBookmark} from "react-icons/fa";
import {BsBookmarkFill} from "react-icons/bs";
import {AiOutlineLeft} from "react-icons/ai";
import Image from "next/image";
import Link from "next/link";
import {setDateLatest} from "../../../lib/commonFn/date";
import {storeInfoState} from "../../../store/reducers/storeInfo/Reducer";
import {SideBarFoldState} from "../../../store/reducers/sideBarFold/Reducer";

interface StoreInfoProps {
  store: storeInfoState;
  postBookMark: (StoreId: number) => Promise<void>;
  deleteBookMark: (StoreId: number) => Promise<void>;
  deleteStoreTab: () => void;
  modalStatus: SideBarFoldState;
  isMobile: boolean;
}

const mainLoader = ({src}: {src: string}) => {
  return `${process.env.NEXT_PUBLIC_IMG_URL}/w_740,h_300${process.env.NEXT_PUBLIC_IMG_ID}/${src}`;
};

const miniLoader = ({src}: {src: string}) => {
  return `${process.env.NEXT_PUBLIC_IMG_URL}/w_200,h_200${process.env.NEXT_PUBLIC_IMG_ID}/${src}`;
};

const StoreInfo = ({
  store,
  postBookMark,
  deleteBookMark,
  deleteStoreTab,
  modalStatus,
  isMobile,
}: StoreInfoProps) => {
  if (store.error) {
    return (
      <section
        className={`${styles.mainBlock} ${
          modalStatus.mobile.searchStoreInfo ? styles.fold : styles.full
        }`}
        data-testid="error"
      >
        <div className={styles.message}>에러가 발생하였습니다</div>
      </section>
    );
  } else if (store.loading) {
    return (
      <section
        className={`${styles.mainBlock} ${
          modalStatus ? styles.fold : styles.full
        }`}
        data-testid="loading"
      >
        <div className={styles.loadingBlock}>
          <Image
            src="/searchResultLoading.gif"
            alt="searchImg"
            layout="fill"
            objectFit="contain"
            priority={true}
          ></Image>
        </div>
      </section>
    );
  } else if (store.content === undefined || modalStatus.desktop.search) {
    return null;
  } else if (store.content?.storeInfo !== undefined) {
    return (
      <section
        className={`${styles.mainBlock} ${
          modalStatus ? styles.fold : styles.full
        }`}
        data-testid="result"
      >
        {store.content.mainPhoto !== undefined ? (
          <div className={styles.mainImgBlock}>
            <Image
              loader={mainLoader}
              src={`/${store.content.mainPhoto}`}
              layout="fill"
              object-fit="cover"
            />
          </div>
        ) : null}
        <div className={styles.titleBlock}>
          {isMobile ? (
            <button
              onClick={deleteStoreTab}
              className={styles.moveList}
              data-testid="deleteStoreTab"
            >
              <AiOutlineLeft size={"25"} />
            </button>
          ) : null}
          <h3 className={styles.storeName}>{store.content.storeInfo.name}</h3>
          <p className={styles.category}>{store.content.storeInfo.category}</p>
          {store.content.bookmark ? (
            <BsBookmarkFill
              className={styles.bookmarkIcon}
              onClick={() => {
                if (store.content?.storeInfo !== undefined) {
                  deleteBookMark(store.content.storeInfo.id);
                }
              }}
            />
          ) : (
            <FaRegBookmark
              className={styles.bookmarkIcon}
              onClick={() => {
                if (store.content?.storeInfo !== undefined) {
                  postBookMark(store.content.storeInfo.id);
                }
              }}
              data-testid="postBookMark"
            />
          )}
        </div>
        <div className={styles.buttonBlock}>
          <button className={styles.contributeButton}>
            <Link href={`/contribute/addreview`}>후기 등록</Link>
          </button>
          <button className={styles.contributeButton}>
            <Link
              href={`/contribute/updatestoreposition/${store.content.storeInfo.id}`}
            >
              위치 수정
            </Link>
          </button>
          <button className={styles.contributeButton}>
            <Link href={`/contribute/updatestoreinfo`}>정보 수정</Link>
          </button>
        </div>
        <div className={styles.infoBlock}>
          <h4 className={styles.subTitle}>정보</h4>
          <p className={styles.infoContentBlock}>
            <BsFillTelephoneFill size={18} className={styles.icon} />
            {store.content.storeInfo.tel === null ||
            store.content.storeInfo.tel === ""
              ? "정보없음"
              : store.content.storeInfo.tel}
          </p>
          <p className={styles.infoContentBlock}>
            <GiPositionMarker size={20} className={styles.icon} />
            {store.content.storeInfo.address === undefined
              ? "정보없음"
              : store.content.storeInfo.address}
          </p>
          <div className={styles.infoContentBlock}>
            <div>
              <BiTimeFive size={20} className={styles.icon} />
            </div>
            {store.content.storeInfo.openingHours === "" ||
            store.content.storeInfo.openingHours === null ? (
              <p>정보없음</p>
            ) : (
              <p className={styles.openingHours}>
                {store.content.storeInfo.openingHours}
              </p>
            )}
          </div>
        </div>
        {store.content.Menus?.length !== 0 ? (
          <div className={styles.menuListBlock}>
            <h4 className={styles.subTitle}>메뉴</h4>
            <div className={styles.menuBlock}>
              {store.content.Menus !== undefined
                ? store.content.Menus.map((src) => {
                    return (
                      <div className={styles.menuImgBlock}>
                        <Image
                          loader={miniLoader}
                          src={`/${src}`}
                          width="105px"
                          height="105px"
                        />
                      </div>
                    );
                  })
                : null}
            </div>
          </div>
        ) : null}
        <div className={styles.reviewBlock}>
          <h4 className={styles.subTitle}>후기</h4>
          <div className={styles.tagListBlock}>
            {store.content.hashtags !== undefined
              ? Object.keys(store.content.hashtags)
                  .sort(function (a, b) {
                    if (store.content?.hashtags !== undefined) {
                      return (
                        store.content.hashtags[b] - store.content.hashtags[a]
                      );
                    } else {
                      return 0;
                    }
                  })
                  .map((tag) => {
                    return (
                      <div className={styles.tag}>
                        #{tag}
                        <span className={styles.tagCount}>
                          {store.content?.hashtags !== undefined
                            ? store.content.hashtags[tag]
                            : null}
                        </span>
                      </div>
                    );
                  })
              : null}
          </div>
          <div className={styles.commentListBlock}>
            {store.content.Reviews?.length !== 0 ? (
              store.content.Reviews?.map((review, index) => {
                return (
                  <div
                    className={styles.commentBlock}
                    key={`${review.content}${index}`}
                  >
                    <div className={styles.commentInfo}>
                      <div className={styles.nickname}>{review.user}</div>
                      <div className={styles.date}>
                        {setDateLatest(review.date)}
                      </div>
                    </div>
                    {review.photos.length !== 0 ? (
                      <div className={styles.imgBlock}>
                        {review.photos.map((src) => {
                          return (
                            <div className={styles.imgs}>
                              <Image
                                loader={miniLoader}
                                src={`/${src}`}
                                width="100px"
                                height="100px"
                              />
                            </div>
                          );
                        })}
                      </div>
                    ) : null}

                    <div className={styles.commentContent}>
                      {review.content}
                    </div>
                    <div className={styles.tagListBlock}>
                      {review.Hashtags.map((hashtagName: string) => {
                        return (
                          <div className={styles.commentTag}>
                            #{hashtagName}
                          </div>
                        );
                      })}
                    </div>
                  </div>
                );
              })
            ) : (
              <p className={styles.noReview}>첫 후기를 등록해 보세요!</p>
            )}
          </div>
        </div>
      </section>
    );
  } else {
    return null;
  }
};

export default StoreInfo;
