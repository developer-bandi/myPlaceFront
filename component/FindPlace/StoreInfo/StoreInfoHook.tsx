import { useDispatch, useSelector } from "react-redux";
import {
  addBookMark,
  deleteBookMark as deleteBookMarkApi,
} from "../../../api/search";
import { RootReducer } from "../../../store";
import {
  initializeStoreInfo,
  setBookmark,
} from "../../../store/reducers/storeInfo/Reducer";

const useStoreInfo = () => {
  const store = useSelector((state: RootReducer) => state.storeInfo);
  const loginedUser = useSelector(
    (state: RootReducer) => state.userLogin.content
  );
  const modalStatus = useSelector((state: RootReducer) => state.searchModal);
  const dispatch = useDispatch();

  const postBookMark = async (storeId: number) => {
    if (!loginedUser) {
      alert("로그인을 해주세요");
    } else {
      try {
        await addBookMark(storeId);
        dispatch(setBookmark(true));
      } catch (err) {
        alert("에러가 발생하였습니다");
      }
    }
  };

  const deleteBookMark = async (storeId: number) => {
    if (!loginedUser) {
      alert("로그인을 해주세요");
    } else {
      try {
        await deleteBookMarkApi(storeId);
        dispatch(setBookmark(false));
      } catch (err) {
        alert("에러가 발생하였습니다");
      }
    }
  };

  const deleteStoreTab = () => {
    dispatch(initializeStoreInfo());
  };

  return {
    store,
    postBookMark,
    deleteBookMark,
    deleteStoreTab,
    modalStatus,
  };
};

export default useStoreInfo;
