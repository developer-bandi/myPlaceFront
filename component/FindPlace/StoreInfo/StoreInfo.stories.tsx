import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import StoreInfo from "./StoreInfo";
import { action } from "@storybook/addon-actions";
import { loader } from "../../../lib/loader";

export default {
  title: "FindPlace/StoreInfo",
  component: StoreInfo,
} as ComponentMeta<typeof StoreInfo>;

const Template: ComponentStory<typeof StoreInfo> = (arg) => {
  return (
    <StoreInfo
      {...arg}
      postBookMark={() =>
        new Promise((resolve) => {
          action("postBookMark");
          resolve();
        })
      }
      deleteBookMark={() =>
        new Promise((resolve) => {
          action("deleteBookMark");
          resolve();
        })
      }
      deleteStoreTab={action("deleteStoreTab")}
    />
  );
};

const makeUrl = loader({ width: 740, height: 300 });

export const error = Template.bind({});
error.args = {
  store: { loading: false, error: true },
  modalStatus: {
    desktop: { search: false, storeInfo: false },
    mobile: { searchStoreInfo: true },
  },
  isMobile: false,
};

export const loading = Template.bind({});
loading.args = {
  store: { loading: true, error: false },
  modalStatus: {
    desktop: { search: false, storeInfo: false },
    mobile: { searchStoreInfo: true },
  },
  isMobile: false,
};

export const normal = Template.bind({});
normal.args = {
  store: {
    content: {
      storeInfo: {
        id: 1,
        name: "상점",
        tel: "051-123-345",
        openingHours:
          "월 : 휴무\n화 : 10:00~19:00\n수 : 10:00~19:00\n목 : 10:00~19:00\n금 : 10:00~19:00\n토 : 10:00~19:00\n일 : 10:00~19:00",
        address: "부산광역시 금정구 장전동",
        category: "카페",
        updatedAt: new Date(),
        latitude: "testLatitude",
        longitude: "testLongitude",
      },
      mainPhoto: makeUrl({
        src: "/place/16625346494021491388606",
      }),
      Reviews: [
        {
          content: "테스트 댓글 입니다",
          user: "place",
          date: "2022-11-11",
          Hashtags: ["깨끗한", "깔끔한"],
          photos: [
            makeUrl({
              src: "/place/16625346494021491388606",
            }),
          ],
        },
      ],
      Menus: [
        makeUrl({
          src: "/place/16625346494021491388606",
        }),
      ],
    },
    loading: false,
    error: false,
  },
  modalStatus: {
    desktop: { search: false, storeInfo: false },
    mobile: { searchStoreInfo: true },
  },
  isMobile: false,
};
