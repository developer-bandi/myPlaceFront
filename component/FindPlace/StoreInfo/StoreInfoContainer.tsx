import { useIsMobile } from "../../../hooks/mediaQuery";
import StoreInfo from "./StoreInfo";
import useStoreInfo from "./StoreInfoHook";

const StoreInfoContainer = () => {
  const { store, postBookMark, deleteBookMark, deleteStoreTab, modalStatus } =
    useStoreInfo();
  const isMobile = useIsMobile();
  return (
    <StoreInfo
      store={store}
      postBookMark={postBookMark}
      deleteBookMark={deleteBookMark}
      deleteStoreTab={deleteStoreTab}
      modalStatus={modalStatus}
      isMobile={isMobile}
    />
  );
};

export default StoreInfoContainer;
