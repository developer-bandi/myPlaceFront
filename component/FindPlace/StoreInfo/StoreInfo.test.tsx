import {
  act,
  fireEvent,
  render,
  renderHook,
  screen,
} from "@testing-library/react";
import axios from "axios";
import {useRouter} from "next/router";
import {Provider} from "react-redux";
import configureMockStore from "redux-mock-store";
import StoreInfo from "./StoreInfo";
import useStoreInfo from "./StoreInfoHook";

jest.mock("axios");
const mockedAxios = axios as jest.Mocked<typeof axios>;

jest.mock("next/router", () => ({
  __esModule: true,
  useRouter: jest.fn(),
}));
const mockRouter = {
  push: jest.fn(),
};
(useRouter as jest.Mock).mockReturnValue(mockRouter);

const alertMock = jest.fn();
window.alert = alertMock;

describe("StoreInfo Presentational 테스트", () => {
  const storeInfoMock = {
    storeInfo: {
      id: 1,
      name: "testName",
      tel: "testTel",
      openingHours: "testOpeningHour",
      address: "testAddress",
      category: "testCategory",
      updatedAt: new Date(),
      latitude: "testLatitude",
      longitude: "testLongitude",
    },
  };
  it("스토어가 렌더링되지 않은 경우", () => {
    const utils = render(
      <StoreInfo
        store={{loading: false, error: false}}
        postBookMark={jest.fn()}
        deleteBookMark={jest.fn()}
        deleteStoreTab={jest.fn()}
        modalStatus={{
          desktop: {search: false, storeInfo: false},
          mobile: {searchStoreInfo: true},
        }}
        isMobile={false}
      />
    );

    expect(utils.container).toMatchSnapshot();
  });
  it("스토어가 로딩중인 경우", () => {
    const utils = render(
      <StoreInfo
        store={{loading: true, error: false}}
        postBookMark={jest.fn()}
        deleteBookMark={jest.fn()}
        deleteStoreTab={jest.fn()}
        modalStatus={{
          desktop: {search: false, storeInfo: false},
          mobile: {searchStoreInfo: true},
        }}
        isMobile={false}
      />
    );
    screen.getByTestId("loading");
    expect(utils.container).toMatchSnapshot();
  });
  it("에러가 발생한 경우", () => {
    const utils = render(
      <StoreInfo
        store={{loading: false, error: true}}
        postBookMark={jest.fn()}
        deleteBookMark={jest.fn()}
        deleteStoreTab={jest.fn()}
        modalStatus={{
          desktop: {search: false, storeInfo: false},
          mobile: {searchStoreInfo: true},
        }}
        isMobile={false}
      />
    );
    screen.getByTestId("error");
    expect(utils.container).toMatchSnapshot();
  });
  it("정상적으로 출력하는 경우", () => {
    const postBookMarkMock = jest.fn();
    const deleteBookMarkMock = jest.fn();
    const utils = render(
      <StoreInfo
        store={{content: storeInfoMock, loading: false, error: false}}
        postBookMark={postBookMarkMock}
        deleteBookMark={jest.fn()}
        deleteStoreTab={deleteBookMarkMock}
        modalStatus={{
          desktop: {search: false, storeInfo: false},
          mobile: {searchStoreInfo: true},
        }}
        isMobile={true}
      />
    );
    screen.getByTestId("result");
    fireEvent.click(screen.getByTestId("postBookMark"));
    fireEvent.click(screen.getByTestId("deleteStoreTab"));
    expect(utils.container).toMatchSnapshot();
    expect(postBookMarkMock).toBeCalledTimes(1);
    expect(deleteBookMarkMock).toBeCalledTimes(1);
  });
});

describe("StoreInfo Hook 테스트", () => {
  let login = false;
  const mockStore = configureMockStore();
  const notLoginMockStore = mockStore({
    userLogin: {},
    storeInfo: {},
    modalStatus: {mobile: {searchStoreInfo: true}},
  });
  const loginMockStore = configureMockStore()({
    userLogin: {content: {id: 1, nickname: "testNickname"}},
    storeInfo: {},
    modalStatus: {mobile: {searchStoreInfo: true}},
  });
  const wrapper = ({children}: {children: React.ReactNode}) => (
    <Provider store={login ? loginMockStore : notLoginMockStore}>
      {children}
    </Provider>
  );
  describe("postBookMark함수 테스트", () => {
    it("로그인 하지 않은 경우", async () => {
      const {result} = renderHook(() => useStoreInfo(), {wrapper});
      await act(async () => {
        result.current.postBookMark(111111);
      });
      expect(alertMock.mock.calls[0][0]).toBe("로그인을 해주세요");
    });
    it("정상적으로 반영된 경우", async () => {
      login = true;
      mockedAxios.post.mockImplementation(() =>
        Promise.resolve({status: 200, data: "test"})
      );
      const {result} = renderHook(() => useStoreInfo(), {wrapper});
      await act(async () => {
        result.current.postBookMark(111111);
      });
      expect(loginMockStore.getActions()[0]).toEqual({
        payload: true,
        type: "storeInfo/setBookmark",
      });
    });
    it("에러가 발생한 경우", async () => {
      mockedAxios.post.mockImplementation(() =>
        Promise.reject({status: 500, data: "error"})
      );
      const {result} = renderHook(() => useStoreInfo(), {wrapper});
      await act(async () => {
        result.current.postBookMark(111111);
      });
      expect(alertMock.mock.calls[1][0]).toBe("에러가 발생하였습니다");
    });
  });

  describe("deleteBookMark함수 테스트", () => {
    it("로그인 하지 않은 경우", async () => {
      login = false;
      const {result} = renderHook(() => useStoreInfo(), {wrapper});
      await act(async () => {
        result.current.deleteBookMark(111111);
      });
      expect(alertMock.mock.calls[2][0]).toBe("로그인을 해주세요");
    });
    it("정상적으로 반영된 경우", async () => {
      login = true;
      mockedAxios.delete.mockImplementation(() =>
        Promise.resolve({status: 200, data: "test"})
      );
      const {result} = renderHook(() => useStoreInfo(), {wrapper});
      await act(async () => {
        result.current.deleteBookMark(111111);
      });
      expect(loginMockStore.getActions()[1]).toEqual({
        payload: false,
        type: "storeInfo/setBookmark",
      });
    });
    it("에러가 발생한 경우", async () => {
      mockedAxios.delete.mockImplementation(() =>
        Promise.reject({status: 500, data: "error"})
      );
      const {result} = renderHook(() => useStoreInfo(), {wrapper});
      await act(async () => {
        result.current.deleteBookMark(111111);
      });
      expect(alertMock.mock.calls[3][0]).toBe("에러가 발생하였습니다");
    });
  });
  it("deleteStoreTab 함수 테스트", () => {
    const {result} = renderHook(() => useStoreInfo(), {wrapper});
    act(() => {
      result.current.deleteStoreTab();
    });
    expect(loginMockStore.getActions()[2]).toEqual({
      payload: undefined,
      type: "storeInfo/initializeStoreInfo",
    });
  });
});
