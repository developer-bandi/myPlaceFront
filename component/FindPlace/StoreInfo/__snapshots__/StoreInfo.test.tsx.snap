// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreInfo Presentational 테스트 스토어가 렌더링되지 않은 경우 1`] = `<div />`;

exports[`StoreInfo Presentational 테스트 스토어가 로딩중인 경우 1`] = `
<div>
  <section
    class="mainBlock fold"
    data-testid="loading"
  >
    <div
      class="loadingBlock"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
      >
        <img
          alt="searchImg"
          data-nimg="fill"
          decoding="async"
          sizes="100vw"
          src="/_next/image?url=%2FsearchResultLoading.gif&w=3840&q=75"
          srcset="/_next/image?url=%2FsearchResultLoading.gif&w=640&q=75 640w, /_next/image?url=%2FsearchResultLoading.gif&w=750&q=75 750w, /_next/image?url=%2FsearchResultLoading.gif&w=828&q=75 828w, /_next/image?url=%2FsearchResultLoading.gif&w=1080&q=75 1080w, /_next/image?url=%2FsearchResultLoading.gif&w=1200&q=75 1200w, /_next/image?url=%2FsearchResultLoading.gif&w=1920&q=75 1920w, /_next/image?url=%2FsearchResultLoading.gif&w=2048&q=75 2048w, /_next/image?url=%2FsearchResultLoading.gif&w=3840&q=75 3840w"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
        />
      </span>
    </div>
  </section>
</div>
`;

exports[`StoreInfo Presentational 테스트 에러가 발생한 경우 1`] = `
<div>
  <section
    class="mainBlock fold"
    data-testid="error"
  >
    <div
      class="message"
    >
      에러가 발생하였습니다
    </div>
  </section>
</div>
`;

exports[`StoreInfo Presentational 테스트 정상적으로 출력하는 경우 1`] = `
<div>
  <section
    class="mainBlock fold"
    data-testid="result"
  >
    <div
      class="titleBlock"
    >
      <button
        class="moveList"
        data-testid="deleteStoreTab"
      >
        <svg
          fill="currentColor"
          height="25"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
          />
        </svg>
      </button>
      <h3
        class="storeName"
      >
        testName
      </h3>
      <p
        class="category"
      >
        testCategory
      </p>
      <svg
        class="bookmarkIcon"
        data-testid="postBookMark"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 384 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M336 0H48C21.49 0 0 21.49 0 48v464l192-112 192 112V48c0-26.51-21.49-48-48-48zm0 428.43l-144-84-144 84V54a6 6 0 0 1 6-6h276c3.314 0 6 2.683 6 5.996V428.43z"
        />
      </svg>
    </div>
    <div
      class="buttonBlock"
    >
      <button
        class="contributeButton"
      >
        <a
          href="/contribute/addreview"
        >
          후기 등록
        </a>
      </button>
      <button
        class="contributeButton"
      >
        <a
          href="/contribute/updatestoreposition/1"
        >
          위치 수정
        </a>
      </button>
      <button
        class="contributeButton"
      >
        <a
          href="/contribute/updatestoreinfo"
        >
          정보 수정
        </a>
      </button>
    </div>
    <div
      class="infoBlock"
    >
      <h4
        class="subTitle"
      >
        정보
      </h4>
      <p
        class="infoContentBlock"
      >
        <svg
          class="icon"
          fill="currentColor"
          height="18"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1.885.511a1.745 1.745 0 0 1 2.61.163L6.29 2.98c.329.423.445.974.315 1.494l-.547 2.19a.678.678 0 0 0 .178.643l2.457 2.457a.678.678 0 0 0 .644.178l2.189-.547a1.745 1.745 0 0 1 1.494.315l2.306 1.794c.829.645.905 1.87.163 2.611l-1.034 1.034c-.74.74-1.846 1.065-2.877.702a18.634 18.634 0 0 1-7.01-4.42 18.634 18.634 0 0 1-4.42-7.009c-.362-1.03-.037-2.137.703-2.877L1.885.511z"
            fill-rule="evenodd"
          />
        </svg>
        testTel
      </p>
      <p
        class="infoContentBlock"
      >
        <svg
          class="icon"
          fill="currentColor"
          height="20"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 17.108c-75.73 0-137.122 61.392-137.122 137.122.055 23.25 6.022 46.107 11.58 56.262L256 494.892l119.982-274.244h-.063c11.27-20.324 17.188-43.18 17.202-66.418C393.122 78.5 331.73 17.108 256 17.108zm0 68.56a68.56 68.56 0 0 1 68.56 68.562A68.56 68.56 0 0 1 256 222.79a68.56 68.56 0 0 1-68.56-68.56A68.56 68.56 0 0 1 256 85.67z"
          />
        </svg>
        testAddress
      </p>
      <div
        class="infoContentBlock"
      >
        <div>
          <svg
            class="icon"
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12 2zm0 18c-4.411 0-8-3.589-8-8s3.589-8 8-8 8 3.589 8 8-3.589 8-8 8z"
            />
            <path
              d="M13 7h-2v5.414l3.293 3.293 1.414-1.414L13 11.586z"
            />
          </svg>
        </div>
        <p
          class="openingHours"
        >
          testOpeningHour
        </p>
      </div>
    </div>
    <div
      class="menuListBlock"
    >
      <h4
        class="subTitle"
      >
        메뉴
      </h4>
      <div
        class="menuBlock"
      />
    </div>
    <div
      class="reviewBlock"
    >
      <h4
        class="subTitle"
      >
        후기
      </h4>
      <div
        class="tagListBlock"
      />
      <div
        class="commentListBlock"
      />
    </div>
  </section>
</div>
`;
